### ISSUE TITLE
Api services for secured portal

### ISSUE THREAD

**Issue by kalletlak** [cBioPortal/cbioportal#1479]
When using secured portal (through some authentication method), if the user doesn't have access to some data-sets and somehow if he gets 'study_id' for those sets and does some queries using the portal api services, then does the user gets success response or should he be responded with unauthorized message?

**Comment by jjgao** [comment]
Thanks @kalletlak.

@n1zea144 I think we should prioritize it. I assume the GSoC project will solve this?

**Comment by n1zea144** [comment]
Now that we are starting to define the new URL/web service calls via the "mini-moonshot" effort, we are starting up conversations with our GSoC student to help define a secured web service - but we are awhile off from having something implemented.

The issue you raise was a bug in our security configuration, but I thought that was fixed.  Can you give me an example URL where this is happening?

For further background, we lockdown URL requests via our spring security config file (look for intercept-url tags):

[applicationContext-security.xml](https://github.com/cBioPortal/cbioportal/blob/master/portal/src/main/resources/applicationContext-security.xml)

Then, servlets that process the webservice should have calls to an AccessControl class we prevents a the user from accessing an unauthorized study:

[WebService.java](https://github.com/cBioPortal/cbioportal/blob/master/core/src/main/java/org/mskcc/cbio/portal/servlet/WebService.java)

It may be that as users added more endpoints, they did not update the applicationContext-security.xml file or did not add a reference to accessControl in the servlet.

In any event, I think we can fix this if there is an issue - unless you want to allow users to access the web service from a client other than a browser.

**Comment by kalletlak** [comment]
Thanks for the information. Those security configurations works. The issue is regarding authorization of studies via web api's.

Consider a case where we have a secured instance and a user has access to limited number of studies. When he access that application he would only query only studies he is authorized to. But when using web api's the study level authorization is not working. Example when the user queries for 'http://<cbio-instance-domain>/api/studies' he would get all the studies from the database even though he is not authorized to access some studies.

Basically in the web api there is no AccessControl class to prevents a the user from accessing an unauthorized study.

**Comment by n1zea144** [comment]
Yeah, seems as though there is no AccessControl reference in the ApiController class.

**Comment by n1zea144** [comment]
@kalletlak @sheridancbio and I assessed the scope of fixing these issues.  Without getting into too much detail, to properly secure the SpringMVC controller that supports the endpoint you provided as an example (basically, anything under /api/*), touches upon a dependency relationship in our maven module/java package structure that needs to be resolved.  Its something we've wanted to do and we will make this a priority now.  I don't know what your timeframe is, but we'll need a couple of weeks to get this working.

**Comment by jjgao** [comment]
Is this still open or solved?

**Comment by kalletlak** [comment]
Its resolved. Thanks
